core/Exception/RedirectException.php

```diff
@@ -28,12 +32,17 @@ public function __construct($message, $redirectTo, $countdown=5)
 
     public function getRedirectionUrl(): string
     {
-        return $this->redirectTo;
+        return $this->isValidRedirect() ? Common::sanitizeInputValue($this->redirectTo) : '';
     }
 
 
     public function getCountdown(): int
     {
         return $this->countdown;
     }
+
+    private function isValidRedirect(): bool
+    {
+        return UrlHelper::isLookLikeUrl($this->redirectTo) && Url::isValidHost(Url::getHostFromUrl($this->redirectTo));
```

Comment:
I'm wondering if we maybe should do similar checks like for redirecting after login as `isLookLikeUrl` does accept some malicious urls.
https://github.com/matomo-org/matomo/blob/d45e58803783ba04082e7d28e087462966cede4a/plugins/Login/Controller.php#L336-L349

Reply:
@sgiehl Thanks for the heads up. I'll integrate some of these into the logic 👍